pkg/envoy/lds/response.go

```diff
@@ -55,13 +65,23 @@ func NewResponse(catalog catalog.MeshCataloger, proxy *envoy.Proxy, _ *xds_disco
 	inboundListener := newInboundListener()
 	if meshFilterChain, err := getInboundInMeshFilterChain(proxyServiceName, cfg); err != nil {
 		log.Error().Err(err).Msgf("Error making in-mesh filter chain for proxy %s", proxy.GetCommonName())
-	} else if meshFilterChain != nil {
+	} else {
+		if !cfg.IsPermissiveTrafficPolicyMode() {
```

Comment:
Another way to ensure filterchain order and also remove indentation level would be to move this outside of the `else` block and use something like this right before `// --- INGRESS -------------------`:
```golang

if !cfg.IsPermissiveTrafficPolicyMode() && len(meshFilterChain.Filters) > 0 {
```

Reply:
We need to add the rbac filter regardless of the other filters in the mesh filter chain.
I could move this out of the existing conditional block, however the ordering is important and needs to be ensured.